plugins/Phaser/PhaserEffect.cpp

```diff
@@ -0,0 +1,684 @@
+			}
+
+			m_currentPeak[i] = qBound(PHA_NOISE_FLOOR, m_currentPeak[i], 10.0f);
+		}
+
+		// Calculate real allpass filter frequencies
+		float lfoResults[2] = {0, 0};
+		if (enableLFO)
+		{
+			m_lfo->tick(&lfoResults[0], &lfoResults[1]);
+		}
+
+		const float realLfo[2] = {lfoResults[0] * amount, lfoResults[1] * amount};
+		const float realInFollow[2] = {m_currentPeak[0] * inFollow, m_currentPeak[1] * inFollow};
+		for (int i = 0; i < 2; i++)
+		{
+			m_displayCutoff[i] = qBound(20.f, detuneWithOctaves(cutoff, realLfo[i] + realInFollow[i]), 20000.f);
+			m_realCutoff[i] = qBound(20.f, detuneWithOctaves(cutoff, (realLfo[i] + realInFollow[i]) * cutoffControl), 20000.f);
+		}
+		
+		//Precalculate the allpass filter coefficients.
+		float apCoeff1[2];
+		float apCoeff2[2];
+		for (int b = 0; b < 2; ++b)
+		{
+			const float w0 = m_twoPiOverSr * m_realCutoff[b];
+			const float a0 = 1 + (sin(w0) / (resonance * 2.f));
+			apCoeff1[b] = (1 - (a0 - 1)) / a0;
+			apCoeff2[b] = (-2*cos(w0)) / a0;
+		}
+
+		float firstAdd[2] = {0, 0};
+		float secondAdd[2] = {0, 0};
+		float delayOut[2] = {0, 0};
+
+		for (int i = 0; i < 2; ++i)
+		{
+			// Calculate delay amount to find read location
+			float readLoc = m_filtFeedbackLoc -
+				qMin(abs(delay + ((realLfo[i] + realInFollow[i]) * delayControl)), 20.f) *
+				0.001f * Engine::audioEngine()->processingSampleRate();
+
+			if (readLoc < 0) {readLoc += m_delayBufSize;}
+			float readLocFrac = fraction(readLoc);
+
+			// Read value from delay buffer
+			if (readLoc < m_delayBufSize - 1)
+			{
+				delayOut[i] = m_filtDelayBuf[i][floor(readLoc)] * (1 - readLocFrac) + m_filtDelayBuf[i][ceil(readLoc)] * readLocFrac;
+			}
+			else// For when the interpolation wraps around to the beginning of the buffer
+			{
+				delayOut[i] = m_filtDelayBuf[i][m_delayBufSize - 1] * (1 - readLocFrac) + m_filtDelayBuf[i][0] * readLocFrac;
+			}
+		}
+
+		float tempAdd[2] = {0, 0};
+		switch (mode)
+		{
+			case 0: case 1:
+			{
+				tempAdd[0] = delayOut[0] * feedback;
+				tempAdd[1] = delayOut[1] * feedback;
+				break;
+			}
+			case 2:
+			{
+				tempAdd[0] = m_lastSecondAdd[0] * feedback;
+				tempAdd[1] = m_lastSecondAdd[1] * feedback;
+				break;
+			}
+			case 3:
+			{
+				tempAdd[0] = delayOut[0] * feedback;
+				tempAdd[1] = delayOut[1] * feedback;
+				for (int a = 0; a < order; ++a)
+				{
+					for (int b = 0; b < 2; ++b)
+					{
+						for (int c = 0; c < 1 + doubleVal; ++c)// Runs twice if doubleVal is true
+						{
+							tempAdd[b] = calcAllpassFilter(tempAdd[b], sample_rate, a, b, apCoeff1[b], apCoeff2[b]);
+						}
+					}
+				}
+			}
+		}
+		if (analog)
+		{
+			for (int i = 0; i < 2; ++i)
+			{
+				// 2x upsample and filter in preparation for distortion
+				float oversampledS1 = m_oversampleAnalogIn1[0]->update(s[i], i) * 2;// * 2 is normalization for 2x oversampling with zero stuffing
+				float oversampledS2 = m_oversampleAnalogIn1[0]->update(0, i);
```

Comment:
Note that the filters for upsampling/downsampling are not linear phase filters, so this will bring extra phase changes.

Reply:
Right.  I haven't been able to find any decent resources for designing linear-phase filters... or, at least, none that I was able to understand at all back when I looked.  I'd really love to learn how though.  Especially if I can manage to design elliptic filters, since those are generally the best to use when it comes to oversampling.  I'm not able to understand all the crazy formulas on the Wikipedia page.

Reply:
The addition of #6553 resolved this issue.  I applied the changes in the latest commit.